runtime/compiler/CMakeLists.txt

```diff
@@ -274,6 +274,17 @@ elseif(OMR_OS_AIX)
 			# Modify the arch tuning value we inherit from omr
 			list(REMOVE_ITEM TR_COMPILE_OPTIONS "-qarch=pwr7")
 			list(APPEND TR_COMPILE_OPTIONS "-qarch=ppc64grsq")
+
+			if(CMAKE_C_COMPILER_IS_XLCLANG)
+				# xlclang/xlclang++ options
+				list(APPEND J9_SHAREDFLAGS -qxlcompatmacros)
+				list(APPEND SPP_FLAGS -qlanglvl=extc99)
+				list(APPEND J9_CXXFLAGS -fno-rtti)
+				set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -lc++")
+			else()
+				# xlc/xlc++ options
+				list(APPEND J9_CFLAGS -qnotempinc -qmbcs)
```

Comment:
These are missing options since cmake was introduced. jdk11 switched to cmake a while back and jdk8 just last week. @zl-wang any comments on the side effects from not using these options?

Reply:
Is adding these to J9_CFLAGS correct? aix-xlc/common.mk adds them to CX_FLAGS.

Reply:
I see CX_FLAGS is used for C_FLAGS, which is different from CXX_FLAGS, so it does seem correct.

Reply:
I think I might want `J9_SHAREDFLAGS` in place of `J9_CFLAGS`, since `CX_FLAGS` is applied to both C/CXX flags and `J9_SHAREDFLAGS` is also applied to J9 C/CXX flags. Going to swap to `J9_SHAREDFLAGS` and test.

Reply:
you meant previously missing those options (-qxlcompatmacros   -qlanglvl  -fno-rtti  and linker -lc++) still allowed the build to be successful?
if yes, then it is perfectly to drop them, except -fno-rtti (which might benefit the ultimate binary size).

Reply:
@zl-wang we're talking about the `-qnotempinc -qmbcs` options used for jdk8 and jdk11 compiles with xlc. The compiles succeed without them, but not sure of the side effects, if there is any problem from not using these options.

Reply:
`J9_SHAREDFLAGS` worked -- pushed changes to switch to this.

Reply:
`-qxlcompatmacros` is provided by eclipse/omr#5794.
`-fno-rtti` is included in `xlc.cmake`.
`-qlanglvl` and `-lc++` are include in `compiler/CMakeLists.txt`.

Reply:
I see `-qnotempinc -qmbcs` specified in omr as well.
https://github.com/eclipse/openj9-omr/blob/openj9/cmake/modules/platform/toolcfg/xlc.cmake#L56-L58

Reply:
@pshipton sorry, on the wrong part previously.  if it can be successful, no implication on that two options.
-qnotempinc  is the default anyway  (meaning:  don't use separate directory for C++ template include headers)
-qmbcs  (meaning:  it can handle multi-byte/double-byte character-set characters in string literals and comments) 
              default is -qnombcs

Reply:
Pushed a new commit which removes some unneeded/duplicated options, per this discussion. Tested with an xlc and an xlclang build -- both passed. I can squash commits after PR testing is complete.

Reply:
@keithc-ca 's comment indicates that -qlanglvl and -lc++ are included in compiler/CMakeLists.txt so aren't needed either. Although I don't find this.

Reply:
See https://github.com/sharon-wang/openj9/blob/xlclang/runtime/compiler/CMakeLists.txt#L280-L281.

Reply:
The `-qlanglvl` and `-lc++` options in `compiler/CMakeLists.txt` are necessary for compilation with xlclang

Reply:
It just confused me that Keith referred to the options being added in this PR along with the others that were not needed.